temporalio/workflow_service.py

```diff
@@ -0,0 +1,280 @@
+            wsv1.RespondWorkflowTaskCompletedRequest,
+            wsv1.RespondWorkflowTaskCompletedResponse,
+        )
+        self.respond_workflow_task_failed = self._new_call(
+            "respond_workflow_task_failed",
+            wsv1.RespondWorkflowTaskFailedRequest,
+            wsv1.RespondWorkflowTaskFailedResponse,
+        )
+        self.scan_workflow_executions = self._new_call(
+            "scan_workflow_executions",
+            wsv1.ScanWorkflowExecutionsRequest,
+            wsv1.ScanWorkflowExecutionsResponse,
+        )
+        self.signal_with_start_workflow_execution = self._new_call(
+            "signal_with_start_workflow_execution",
+            wsv1.SignalWithStartWorkflowExecutionRequest,
+            wsv1.SignalWithStartWorkflowExecutionResponse,
+        )
+        self.signal_workflow_execution = self._new_call(
+            "signal_workflow_execution",
+            wsv1.SignalWorkflowExecutionRequest,
+            wsv1.SignalWorkflowExecutionResponse,
+        )
+        self.start_workflow_execution = self._new_call(
+            "start_workflow_execution",
+            wsv1.StartWorkflowExecutionRequest,
+            wsv1.StartWorkflowExecutionResponse,
+        )
+        self.terminate_workflow_execution = self._new_call(
+            "terminate_workflow_execution",
+            wsv1.TerminateWorkflowExecutionRequest,
+            wsv1.TerminateWorkflowExecutionResponse,
+        )
+        self.update_namespace = self._new_call(
+            "update_namespace",
+            wsv1.UpdateNamespaceRequest,
+            wsv1.UpdateNamespaceResponse,
+        )
+
+    @abstractmethod
+    async def _rpc_call(
+        self,
+        rpc: str,
+        req: google.protobuf.message.Message,
+        resp_type: Type[WorkflowServiceResponse],
+        *,
+        retry: bool = False,
+    ) -> WorkflowServiceResponse:
+        raise NotImplementedError
+
+    def _new_call(
+        self,
+        name: str,
+        req_type: Type[WorkflowServiceRequest],
+        resp_type: Type[WorkflowServiceResponse],
+    ) -> "WorkflowServiceCall[WorkflowServiceRequest, WorkflowServiceResponse]":
+        return WorkflowServiceCall(self, name, req_type, resp_type)
+
+
+class WorkflowServiceCall(Generic[WorkflowServiceRequest, WorkflowServiceResponse]):
+    def __init__(
+        self,
+        service: WorkflowService,
+        name: str,
+        req_type: Type[WorkflowServiceRequest],
+        resp_type: Type[WorkflowServiceResponse],
+    ) -> None:
+        self.service = service
+        self.name = name
+        self.resp_type = resp_type
+
+    async def __call__(
+        self, req: WorkflowServiceRequest, *, retry: bool = False
+    ) -> WorkflowServiceResponse:
+        return await self.service._rpc_call(self.name, req, self.resp_type, retry=retry)
+
+
+class BridgeWorkflowService(WorkflowService):
+    @staticmethod
+    async def connect(target_url: str) -> "BridgeWorkflowService":
```

Comment:
I of course need to take more params here and tunnel them through. I only had `target_url` to get off the ground but I obviously need to add the rest now.

Reply:
Done